### DISCUSSION TITLE
What is cooking?

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1127] [Team Posts]
Hey Tim and Arturo!

As I'm coming out of my annual reverse hibernation (I mean most do hibernation when it's cold...) and started doing some MusE work I wanted to ask what is brewing?

You did a bigger commit a while ago Tim, I'm going through the Changelog and trying to grasp what it means. As with much "engine"-work, the implications aren't immediately apparent.

For myself, apart from the dozens of ideas that hide in dusty corners that, may or may not become anything. I think the thing I would like to get working is mostly vst3. Not sure I'm the man to do it but I'd like to spend some time on it. 
Unless I'm mistaken you did some initial work with vst3, Tim? Or did you do a lot already? :) Anything to share?

**Comment by terminator356** [comment]
So I have three branches in the pipeline, from most to least recent:
1) Someone requested plugin 'soft bypass' support. I have added that.
At the same time I noticed that our support for logarithmic controllers is **broken**.
We actually have no support for log plugin controls! For example LV2 plugins.
All of the above resulted in some really substantial thorough changes and I
 must say this time it will be 'right' and clean.

2) @donarturo11 showed a crash bug with controllers and proposed a fix.
But I said I would take care of it. I am still working on that fix. 
My apologies for the delay there.

3) And finally, remember that whole 'hidden wave events' thing from two years ago 
 where you came up with a brilliant solution of using the wave offset (sPos)
 to automatically resize the wave when resizing wave part borders?
Yeah I did a lot of major work on that and it is almost finished.
I took your idea and ran with it and it works! The changes are huge!

VST3: Yeah I have been examining that quite closely recently, while researching
 the 'soft bypass' request above. I downloaded the SDK again and I'm looking at it.
But still nothing at all has been added or done to prepare - except for 
 a few things related to 'soft bypass' - I 'left the door open' for when
 we eventually add VST3. (VST2 has only a soft bypass function call, while VST3 supports
 an actual soft bypass controller just like LV2.)

**Comment by StaffanMelin** [comment]
I don't feel that I have a log of weight here :) but I also think that VST3 support is a major issue for keeping MusE up to date. So if any of you could look into this, I think it will keep MusE fresh.

I saw that (even) QTractor has added CLAP support. Is that something that could be considered at the same time?

**Comment by terminator356** [comment]
Never heard of CLAP. I have just browsed the introduction.

I thought that you might have been referring to GMPI,
 which was (is) a similar effort at making an open source API:
[https://en.wikipedia.org/wiki/Generalized_Music_Plug-in_Interface](url)

Looks like CLAP might actually take hold though, unlike GMPI...

**Comment by StaffanMelin** [comment]
And I had never heard of GMPI! :)

This thread is a bit interesting:
https://linuxmusicians.com/viewtopic.php?t=25105

**Comment by spamatica** [comment]
Hey Staffan, sorry I forgot you are here too, welcome! :) I saw you dropped your winter-album on linuxmusicians \o/.

Regarding CLAP, I heard of it a few weeks ago. My initial feeling is that it may go somewhere (without knowing too much about it)! But I think it is that is too early to jump on that train. Better use our limited resources where the usefulness is greater. 
But then again, I've always said that the one coding is king, what gets implemented is up to the person at the keyboard.

Regarding VST3, it's great you are already started poking at it Tim! If you have some code I encourage you to publish a branch with it, possibly I could be of help. =)
